client/my-sites/email/email-providers-stacked-comparison/provider-cards/price-badge.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { FunctionComponent, ReactElement } from 'react';
+import PromoCardPrice from 'calypso/components/promo-section/promo-card/price';
+
+type PriceBadgeProps = {
+	additionalPriceInformationComponent?: ReactElement | null;
+	className: string;
+	priceComponent: ReactElement;
+};
+
+const PriceBadge: FunctionComponent< PriceBadgeProps > = ( props ) => {
+	const { additionalPriceInformationComponent, className, priceComponent } = props;
+
+	const priceBadgeClass = `${ className }__price-badge`;
+
+	const additionalPriceInformationClass = `${ className }__provider-additional-price-information`;
```

Comment:
I'm going to leave this as it-is (same thing for the `PriceWithInterval` component) as this may just generate more conflicts when https://github.com/Automattic/wp-calypso/pull/58841 and https://github.com/Automattic/wp-calypso/pull/58843 are merged (they both depend on this pull request). However I don't think we should construct class names dynamically like this.

First, it makes it very difficult to actually look up that class in our code, and see if it is used. It also doesn't help with code duplication - see `google-workspace-card.scss` and `email-provider-stacked-card__title-price-badge.scss` which contain the same styles. It would be better to define a `price_badge` class, and allow this component to accept an optional class name that could be used to tweak it for a specific email provider (if that's really needed, something I'm not sure of).